### ISSUE TITLE
get "perfect match" on a series

### ISSUE THREAD

**Issue by ramarro123** [ta4j/ta4j#721]
it will be great to have a way to check a serie of historical data to extract the best strategy based on brute force, iterate over the BarSeries finding the best buy/sell series match for all the sequence.

it's a feature you think worthy or there is a way to extract this data?

**Comment by MarcDahlem** [comment]
I created a rudimentary strategy "Ta4jOptimalTradingStrategy" in my playground here: https://github.com/MarcDahlem/bxbot/blob/feature/ta4j-backtesting/bxbot-exchanges/src/main/java/com/gazbert/bxbot/exchanges/ta4jhelper/Ta4jOptimalTradingStrategy.java

Not sure if it completely works an all edge-cases, but it shows really good trading results :D

If the team @team172011 wants such a perfect match strategy in their repo, I could create an according PR here.

(While working with this optimal strategy, I think I found a bug with the cost models. The backtest executions of Gains and Losses result in completely different values than I expected and did not match the expected results I calculated.

I will write a testcase for this for reproduction purposes and will open another Issue ticket if the test will fail)
